core/kazoo_amqp/src/gen_listener.erl

```diff
@@ -776,74 +796,112 @@ format_status(_Opt, [_PDict, #state{module=Module
     end.
 
--spec distribute_event(callback_data(), kz_json:object(), basic_deliver(), state()) -> state().
-distribute_event(CallbackData, JObj, BasicDeliver, #state{responders=Responders
-                                                         ,consumer_key=ConsumerKey
-                                                         }=State) ->
+-spec distribute_event(callback_data(), kz_json:object(), deliver(), state()) -> state().
+distribute_event(CallbackData, JObj, Deliver, #state{responders=Responders
+                                                    ,consumer_key=ConsumerKey
+                                                    }=State) ->
     Key = kz_util:get_event_type(JObj),
-    _ = [kz_util:spawn(fun client_handle_event/5, [JObj
+    Channel = kz_amqp_channel:consumer_channel(),
+    _ = [kz_util:spawn(fun client_handle_event/6, [JObj
+                                                  ,Channel
                                                   ,ConsumerKey
                                                   ,Callback
                                                   ,CallbackData
-                                                  ,BasicDeliver
+                                                  ,Deliver
                                                   ])
          || {Evt, Callback} <- Responders,
             maybe_event_matches_key(Key, Evt)
         ],
     State.
 
--spec client_handle_event(kz_json:object(), kz_amqp_channel:consumer_pid(), responder_callback(), callback_data(), basic_deliver()) -> any().
-client_handle_event(JObj, ConsumerKey, Callback, CallbackData, BasicDeliver) ->
+-spec client_handle_event(kz_json:object(), kz_amqp_channel:consumer_channel(), kz_amqp_channel:consumer_pid(), responder_mfa(), callback_data(), deliver()) -> any().
+client_handle_event(JObj, 'undefined', ConsumerKey, Callback, CallbackData, Deliver) ->
     _ = kz_util:put_callid(JObj),
     _ = kz_amqp_channel:consumer_pid(ConsumerKey),
-    client_handle_event(JObj, Callback, CallbackData, BasicDeliver).
-
--spec client_handle_event(kz_json:object(), responder_callback(), callback_data(), basic_deliver()) -> any().
-client_handle_event(JObj, Fun, CallbackData, BasicDeliver)
-  when is_function(Fun, 3) -> Fun(JObj, CallbackData, BasicDeliver);
-client_handle_event(JObj, Fun, CallbackData, _BasicDeliver)
-  when is_function(Fun, 2) -> Fun(JObj, CallbackData);
-client_handle_event(JObj, {Module, Fun}, CallbackData, BasicDeliver) ->
-    case erlang:function_exported(Module, Fun, 3) of
-        'true' -> Module:Fun(JObj, CallbackData, BasicDeliver);
-        'false' -> Module:Fun(JObj, CallbackData)
-    end.
-
--spec callback_handle_event(kz_json:object(), basic_deliver(), state()) ->
+    client_handle_event(JObj, Callback, CallbackData, Deliver);
+client_handle_event(JObj, Channel, ConsumerKey, Callback, CallbackData, Deliver) ->
+    _ = kz_util:put_callid(JObj),
+    _ = case is_pid(Channel)
```

Comment:
Since the previous clause handles `Channel = 'undefined'`, can't you skip this `is_pid(Channel)` check? Then you could do:

```erlang
kz_amqp_channel:consumer_pid(ConsumerKey),
is_process_alive(Channel)
    andalso kz_amqp_channel:consumer_channel(Channel),
client_handle_event(...)
```

Reply:
yep, left over from previously handling. fixing